Update enable_i2c.md (#17005)
home-assistant/home-assistant.github.io: source/_includes/common-tasks/enable_i2c.md

## Enable I2C

Home Assistant using the Home Assistant Operating System which is a managed environment, which means you can't use existing methods to enable the I2C bus on a Raspberry Pi. In order to use I2C devices you will have to 
- Enable I2C for the Home Assistant Operating System 

> - Configure the Raspberry Hats binary sensor integration to enable the I2C integration

Reviewer: This is only required if you use the I2C hat. I2C can also be used without that hat, so I would just drop this line

```suggestion
```

- Setup I2C sensors

> ### Step by step instructions to enable I2C for the Home Assistant Operating System

Reviewer 2: This page is about the Home Assistant Operating System there is no need to put that in a big header again. Please revert this line change.

> ### Step by step instructions to enable I2C via SD card

Reviewer: ```suggestion
### Enable I2C with an SD card reader
```

> #### Use SD Card/Step 1 - Access the Home Assistant Operating System boot partition

Reviewer 2: This replacing of the header, removed the introduction paragraph.

Author: Section titles cleaned of SD card reference

You will need:
- SD card reader
- SD card with Home Assistant Operating System flashed on it

Shutdown/turn-off your Home Assistant installation and unplug the SD card.
Plug the SD card into an SD card reader and find a drive/file system named
`hassos-boot`. The file system might be shown/mounted automatically. If not,
use your operating systems disk management utility to find the SD card reader
and make sure the first partition is available.

> #### Use SD Card/Step 2 - Add files to enable I2C

Reviewer 2: Let's not add "Use SD Card" to each header, instead, make the introduction correct.

Author: Removed SD card reference

  dtparam=i2c_arm=on
  ```

#### Start with the new OS configuration

- Insert the SD card back into your Raspberry Pi.
- On startup, the `hassos-config.service` will automatically pickup the new
  `rpi-i2c.conf` configuration.
- Another reboot might be necessary to make sure the just imported `rpi-i2c.conf` is
  present at boot time.

> ### Via Home Assistant Operating System Terminal

Reviewer: ```suggestion
### Enable I2C via Home Assistant Operating System Terminal
```

  dtparam=i2c_arm=on
  ```

#### Use SD Card/Step 3 - Start with the new OS configuration

- Insert the SD card back into your Raspberry Pi.
- On startup, the `hassos-config.service` will automatically pickup the new
  `rpi-i2c.conf` configuration.
- Another reboot might be necessary to make sure the just imported `rpi-i2c.conf` is
  present at boot time.

> ### Alternative to SD Card via Home Assistant Operating System Terminal

Reviewer 2: Could you explain why this title needed changing?

Author: Shortend the section title

  sync
  reboot
  ```
#### Check I2C activation on OS level

After the reboot of the host, the I2C activation should now be visible under /dev. You may check by using the add-on `Terminal & SSH` and list the directory `/dev`. If `ls` shows an entry `i2c-1` or similar, your I2C config might work. You can check the status of I2C kernel modules by entering `lsmod | grep i2c`. If they are loaded, you should find at least the entry `i2c_dev`. Active usage of the modules is indicated by a number, e.g. `i2c_dev 20480 2` would indicate two active I2C data sources (e.g. a BMP280 temperature & pressure sensor) 

> An active I2C can also be check with a multi meter showing 3.3 V on the I2C pins GPIO2 and GPIO3. 

Reviewer 2: This section does not apply to a regular installation and requires third party add-ons. It should therefore not be added here.

Author: Removed the reference to third party add on. Kept lsmod and more explicit description of the dev folder

#### Configuration of the Home Assistant to enable a BME280 sensor 
  
Connect a I2C device to the GPIO2 (SDA1), GPIO3 (SCL) and VCC/GND as needed. Ensure that this wiring is working (e.g by using a Raspberry OS with configured I2C) and note the I2C address of the  device. For a BMP280 this might be 0x76.  

Edit the `configuration.yaml` e.g. via the add-on `File Editor`. You need to activate the  binary sensor plattform `raspihats` to enable I2C for the Home Assistant core. The I2C sensor needs and additional entry. A working entry may look like this:

```shell
# Enable Raspihats
binary_sensor:
  - platform: raspihats

# Enable BMP 280 sensor
sensor:
  - platform: bmp280
    i2c_address: 0x76
```

After a reboot of host, you can find the sensor listed in `Developer Tools` in the tab `States`as `sensor.bmp280_temperature`and `sensor.bmp280_pressure`. 

>   

Reviewer 2: This is not related to this page. Please remove that. This belongs on a integration page. Not in the common tasks.

Author: Removed the section as requested

  sync
  reboot
  ```

> ### Check I2C activation on OS level

Reviewer: ```suggestion
### Troubleshooting
```

> After the reboot of the host, the I2C activation should now be visible under /dev.  If the folder contains an entry `i2c-1` or similar, your I2C config might work. You can check the status of I2C kernel modules by entering `lsmod | grep i2c`. If they are loaded, you should find at least the entry `i2c_dev`. Active usage of the modules is indicated by a number, e.g. `i2c_dev 20480 2` would indicate two active I2C data sources (e.g. a BMP280 temperature & pressure sensor) 

Reviewer: ```suggestion
After rebooting the host there should be `i2c-0` and similar device files in `/dev`. If such device files are missing, enabling I2C failed for some reason. You can check the status of I2C kernel modules by using `lsmod | grep i2c` in the terminal. If they are loaded, you should find at least the entry `i2c_dev`. Active usage of the modules is indicated by a number, e.g. `i2c_dev 20480 2` would indicate two active I2C device files.
```
